src/server/lib/recaptcha.ts

```diff
@@ -1,74 +1,87 @@
 import { NextFunction, Request, Response } from 'express';
-import { RecaptchaV2 } from 'express-recaptcha';
 import { CaptchaErrors } from '@/shared/model/Errors';
 import { getConfiguration } from '@/server/lib/getConfiguration';
 import { logger } from '@/server/lib/serverSideLogger';
 import { trackMetric } from '@/server/lib/trackMetric';
 import { HttpError } from '@/server/models/Error';
-import { featureSwitches } from '@/shared/lib/featureSwitches';
+import { fetch } from '@/server/lib/fetch';
 
 const {
-  googleRecaptcha: { secretKey, siteKey },
-  stage,
+  googleRecaptcha: { secretKey },
 } = getConfiguration();
 
-const recaptcha = new RecaptchaV2(siteKey, secretKey);
-
 const recaptchaError = new HttpError({
   message: CaptchaErrors.GENERIC,
   status: 400,
   name: 'RecaptchaError',
   code: 'EBADRECAPTCHA',
 });
 
+type RecaptchaErrorCode =
+  | 'missing-input-secret'
+  | 'invalid-input-secret'
+  | 'missing-input-response'
+  | 'invalid-input-response'
+  | 'bad-request'
+  | 'timeout-or-duplicate';
+
+interface RecaptchaAPIResponse {
+  success: boolean;
+  challenge_ts: string;
+  hostname: string;
+  'error-codes'?: RecaptchaErrorCode[];
+}
+
 /**
- * Throws a generic error if the recaptcha check has failed.
- * @param recaptchaResponse
+ * Takes the reCAPTCHA token from the client and sends it to the Google reCAPTCHA API to verify it.
+ * If the token is valid, the request will be allowed to continue; if not, an error will be thrown.
+ * Documentation: https://developers.google.com/recaptcha/docs/verify
+ * @param req Express request object
+ * @param _ Express response object (not used)
+ * @param next Express next function
+ * @returns Throws an error if the reCAPTCHA check has failed; otherwise, calls the next function
  */
-const checkRecaptchaError = (req: Request, _: Response, next: NextFunction) => {
-  if (!req.recaptcha) {
+const handleRecaptcha = async (
+  req: Request,
+  _: Response,
+  next: NextFunction,
+): Promise<void> => {
+  const recaptchaToken = req.body['g-recaptcha-response'];
+  // If the recaptcha response is missing entirely, throw an error which will be shown to the user.
+  if (!recaptchaToken) {
     return next(recaptchaError);
   }
+  const recaptchaResponse = await fetch(
+    'https://www.google.com/recaptcha/api/siteverify',
+    {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: `secret=${secretKey}&response=${recaptchaToken}`,
+    },
+  );
+  const recaptchaResponseJson =
+    (await recaptchaResponse.json()) as RecaptchaAPIResponse;
```

Comment:
my only suggestion would be to wrap any `await` calls in a `try`/`catch` block, in case there are any runtime errors, e.g. if the fetch call fails/can't parse the json etc.